src/deepsparse/utils/state.py

```diff
@@ -62,3 +92,24 @@ def update_value(self, attribute: str, value: Union[str, int, list]):
 
     def update_state(self, value: Any):
         self._current_state.update(value)
+
+    def get_state(self, key):
+        if key in self.current_state:
+            return self.current_state[key]
+
+    def copy_state(self, props=["timer"]):
```

Comment:
Won't a shallow copy have the same effect as this function?

Reply:
Every InferenceState may have their own context, so you can't share it with other Operator.
The only thing we share is the Timer.